docs/aks/mdmonboarding/mdm_onboarding.ps1

```diff
@@ -0,0 +1,208 @@
+﻿<# 
+    .DESCRIPTION 
+		Adds the Monitoring Metrics Publisher role assignment to the specified AKS cluster        
+      
+	
+    .PARAMETER SubscriptionId
+        Subscription Id that the AKS cluster is in
+
+    .PARAMETER ResourceGroupName
+        Resource Group name that the AKS cluster is in
+            
+	 .PARAMETER clusterName
+        Name of the AKS cluster.
+#>
+
+param(
+	[Parameter(mandatory=$true)]
+	[string]$SubscriptionId,
+	[Parameter(mandatory=$true)]
+	[string]$ResourceGroupName,
+	[Parameter(mandatory=$true)]
+	[string] $clusterName
+)
+
+
+# checks the required Powershell modules exist and if not exists, request the user permission to install
+$azAccountModule = Get-Module -ListAvailable -Name Az.Accounts
+$azAksModule = Get-Module -ListAvailable -Name Az.Aks 
+$azResourcesModule = Get-Module -ListAvailable -Name Az.Resources
+
+if (($azAccountModule -eq $null) -or ($azAksModule -eq $null) -or ($azResourcesModule -eq $nul)) {
+
+
+    $currentPrincipal = New-Object Security.Principal.WindowsPrincipal([Security.Principal.WindowsIdentity]::GetCurrent())
+
+    if ($currentPrincipal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
+	    Write-Host("Running script as an admin...")
+	    Write-Host("")
+    } else {
+	    Write-Host("Please run the script as an administrator") -ForegroundColor Red
+	    Stop-Transcript
+	    exit
+    }
+
+
+    $message = "This script will try to install the latest versions of the following Modules : `
+			    Az.Resources, Az.Accounts and Az.Aks using the command`
+			    `'Install-Module {Insert Module Name} -Repository PSGallery -Force -AllowClobber -ErrorAction Stop -WarningAction Stop'
+			    `If you do not have the latest version of these Modules, this troubleshooting script may not run."
+    $question = "Do you want to Install the modules and run the script or just run the script?"
+
+    $choices = New-Object Collections.ObjectModel.Collection[Management.Automation.Host.ChoiceDescription]
+    $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&Yes, Install and run'))
+    $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&Continue without installing the Module'))
+    $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&Quit'))
+
+    $decision = $Host.UI.PromptForChoice($message, $question, $choices, 0)
+
+    switch ($decision) {
+	    0 { 
+
+           try {
+			    Write-Host("Installing Az.Resources...")
```

Comment:
A check here to install the module only if it is required.
if ($azAccountModule -eq $null) {
// install 
}

Reply:
done.